relay-server/src/utils/rate_limits.rs

```diff
@@ -458,6 +462,12 @@ impl Enforcement {
             return false;
         }
 
+        if item.ty() == &ItemType::Transaction && self.spans_indexed.is_active() {
+            // We cannot remove nested spans from the transaction, but we can prevent them
+            // from being extracted into standalone spans.
+            item.set_spans_extracted(true);
```

Comment:
Seems quite hacky but I can't think of anything better.

Reply:
We could rename the flag to `skip_span_extraction`. Would that help?